app/src/main/kotlin/com/waz/zclient/devices/domain/GetAllClientsUseCase.kt

```
@@ -0,0 +1,20 @@
+package com.waz.zclient.devices.domain
+
+import com.waz.zclient.core.resources.Resource
+import com.waz.zclient.core.usecase.coroutines.UseCase
+import com.waz.zclient.devices.data.ClientsRepository
+import com.waz.zclient.devices.domain.model.Client
+import timber.log.Timber
+
+class GetAllClientsUseCase(private val clientsRepository: ClientsRepository)
+    : UseCase<List<Client>, Unit>() {
+
+    override suspend fun run(params: Unit): Resource<List<Client>> {
+        return try {
```

We should avoid using this try catch as much as possible, that is why I was suggesting the usage of ```Either<L, R>```.

We may end up eating the exception if the caller is not aware of the possible check. This approach is very prone to "hard to find bugs" since we loose awareness but let's open a discussion: 
https://softwareengineering.stackexchange.com/questions/365427/try-catch-log-rethrow-is-anti-pattern